### PR TITLE
Fix broken platform tests

### PR DESCRIPTION
**Pull request by gwenkidd**
# Description
Fixed the broken tests within these files: 
test_basehistorian.py
test_connection.py
test_instance_setup.py
test_keystore.py
test_platform_rmq.py
test_platform_web.py

Each commit focuses on a different test file with issues linked in the commit messages that describe what the problem with the tests were. 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Run tests.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: volttrontesting/fixtures/volttron_platform_fixtures.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def volttron_instance_rmq(request):
                 params=[
                     dict(messagebus='zmq', ssl_auth=False),
                     pytest.param(dict(messagebus='zmq', ssl_auth=True), marks=ci_skipif),
-                    pytest.param(dict(messagebus='rmq', ssl_auth=True), marks=rmq_skipif),
+                    #pytest.param(dict(messagebus='rmq', ssl_auth=True), marks=rmq_skipif),
```
Comment (craig8) [inline volttrontesting/fixtures/volttron_platform_fixtures.py:253]: This needs to be runable...unless you were given other instructions.

File: volttrontesting/platform/test_rmq_reconnect.py [hunk 2]
```diff
@@ -48,7 +48,7 @@
 from volttron.utils.rmq_config_params import RMQConfig
 from volttron.platform.vip.agent.errors import Unreachable
 
-pytestmark = [pytest.mark.xfail]
+#pytestmark = [pytest.mark.xfail]
```
Comment (craig8) [inline volttrontesting/platform/test_rmq_reconnect.py:51]: just remove this

File: volttrontesting/platform/test_rmq_reconnect.py [hunk 3]
```diff
@@ -162,7 +164,7 @@ def test_rmq_reconnect_with_publish(volttron_instance_rmq, publisher_agent, subs
     gevent.sleep(0.1)
     assert subscriber_agent.callback.call_count >= 2
 
-
+# PASSING
```
Comment (craig8) [inline volttrontesting/platform/test_rmq_reconnect.py:167]: great but probably don't need to comment it :)

File: volttrontesting/fixtures/volttron_platform_fixtures.py (hunk 1 above)
Comment (gwenkidd) [reply volttrontesting/fixtures/volttron_platform_fixtures.py:253]: Chandrika and Shwetha told me to comment that out while they work on the RMQ issue.

**Comment by shwethanidd** [comment]
Fixes #2878, #2879, #2880, #2884

**Pull request merged**